scripts/fill_dcp1_file_metadata/fill_dcp1_metadata.py

```diff
@@ -0,0 +1,249 @@
+
+    return uuid_metadata_map_filtered
+
+
+def set_up_ingestapi(ingest_token: str, ingest_base_url: str) -> IngestApi:
+    """
+    Set up ingestApi object, with credentials to access and modify information in the database. URL is not asked because
+    this script can only be run in production (Azul service not available in dev/staging)
+
+    :param ingest_token:    Token to access/modify entities in ingest
+    :param ingest_base_url: Ingest's API base URL. Defaults to prod
+    :return ingest_api:     IngestApi object set up with token
+    """
+    ingest_api = IngestApi(ingest_base_url)
+    ingest_api.set_token(f"Bearer {ingest_token}")
+    return ingest_api
+
+
+def patch_file_metadata(uuid_metadata_map: dict, ingest_api: IngestApi, dry_run: bool) -> [list, list]:
+    """
+    Send a PUT request to each of the files in the uuid_size_map with the updated size and fileContentType extracted
+    from Azul
+    :param uuid_metadata_map:   {file_uuid: metadata} map of all the files that need to be updated
+    :param ingest_api:          IngestApi object with headers set up
+    :param dry_run:             bool value, if True no PUT operations will be performed (But information will be logged)
+    :return not_found:          List of file metadata that was not found in ingest
+    :return already_filled:     List of file metadata that was already filled
+    """
+    not_found = []
+    already_filled = []
+    for uuid, metadata in tqdm.tqdm(uuid_metadata_map.items()):
+        print(uuid, metadata)
+        # Check file exists in Ingest (Non-organically described matrices are virtually indistinguishable via metadata)
+        try:
+            logging.info(f"Searching for file metadata for file {metadata['name']}")
+            entity = ingest_api.get_entity_by_uuid('files', uuid)
+        except HTTPError:
+            not_found.append(metadata)
+            continue
+
+        # Check entity is not already filled in - Script will only fill size and fileContentType when both are missing
+        if entity.get('size') and entity.get('fileContentType'):
+            already_filled.append(metadata)
+            continue
+
+        # Add size and fileContentType to metadata
+        entity['size'] = metadata['size']
+        data_type = f"{'application/gzip' if metadata['format'] == 'fastq.gz' else 'application/octet-stream'}; dcp-type=data"
+        entity['fileContentType'] = data_type
+
+        logging.info(f"PATCH file {entity.get('fileName')}, uuid {uuid}, with size {entity['size']} and dcp-type {entity['fileContentType']}")
+
+        if dry_run:
+            continue
+
+        # PUT the metadata back with the newly introduced values
+        ingest_api.put(entity['_links']['self']['href'], json=entity)
+        logging.info(f"PATCHED file {entity.get('fileName')}, url: {entity['_links']['self']['href']}")
+
+    return not_found, already_filled
```

Comment:
Generally speaking, it is more readable and maintainable to return a namedtuple or dataclass rather than a tuple.